js/post/update_post.js

```diff
@@ -368,3 +257,52 @@ function updateNavGraph() {
         nav_graph.innerHTML = 'Check My POSt! (' + getCookie('activecount') + '/4.0)';
     } 
 }
+
+
+/**
+ * Updates list of current active (selected) courses in Graph.
+**/
+function updateActiveCourses() {
+    'use strict';
+
+    activeCourses = [];
+
+    // check for active CSC courses
+    for (var areaName in areas) {
+        if (areas.hasOwnProperty(areaName)) {
+            for (var i = 0; i < areas[areaName].length; i++) {
+                if ((getCookie(areas[areaName][i]) === 'active' || getCookie(areas[areaName][i]) === 'overridden') && 
+                    areas[areaName][i] != 'CSC104') {
+                    activeCourses.push(areas[areaName][i]);
+                }
+            }
+        }
+    }
+
+    // check for active math courses
+    for (var i = 0; i < math.length; i++) {
+        if (getCookie(math[i]) === 'active') {
+            activeCourses.push(math[i]);
+        }
+    }
+}
+
+
+function updateMatCreditCount() {
+    var spec_extra = $('#specextra')[0].getElementsByTagName('input');
+    var maj_extra = $('#majextra')[0].getElementsByTagName('input');
+
+    for (var k = 0; k < 4; k++) {
+        if (spec_extra[k].value.indexOf('MAT') > -1 || spec_extra[k].value.indexOf('STA') > -1) {
+            specialist.creditCount += 0.5;
+            specialist.filledTextboxesExtra += 1;
```

Comment:
Would it be better to make more formal objects with their own methods, just like how `Edge`, `Node`, `Course`, and `Section` are defined now?